addons/sale_project/models/sale_order_line.py

```diff
@@ -137,12 +137,24 @@ def _compute_qty_delivered(self):
     def _compute_analytic_distribution(self):
         super()._compute_analytic_distribution()
         for line in self:
-            if line.display_type or line.analytic_distribution or not line.product_id:
-                continue
             project = line.product_id.project_id or line.order_id.project_id
-            distribution = project._get_analytic_distribution()
-            if distribution:
-                line.analytic_distribution = distribution
+            if line.display_type or not line.product_id or not project:
+                continue
+
+            if line.analytic_distribution:
+                applied_root_plans = self.env['account.analytic.account'].browse(
+                    list({int(account_id) for ids in line.analytic_distribution for account_id in ids.split(",")})
+                ).mapped('root_plan_id')
+                if project_analytic_accounts := project._get_analytic_accounts():
+                    accounts_to_add = self.env['account.analytic.account']
+                    analytic_account_2_root_plan = {account: account.root_plan_id for account in project_analytic_accounts}
+                    for account, root_plan in analytic_account_2_root_plan.items():
+                        if not root_plan & applied_root_plans:
+                            accounts_to_add |= account
+                    if accounts_to_add:
```

Comment:
I feel like you are over complicating it, or didn't I understand what you are trying to do?
```suggestion
                if accounts_to_add := project._get_analytic_accounts().filtered(
                    lambda account: account.root_plan_id not in applied_root_plans
                ):
```

Reply:
I was definitely hyper complicating it! Thank you :)